lib/OpenQA/Schema/ResultSet/Jobs.pm

```diff
@@ -94,10 +94,9 @@ MACHINE. For each set of dupes, only the latest job found is included in
 the return array.
 
 =cut
-sub latest_jobs {
-    my ($self, $until) = @_;
-
-    my @jobs = $self->search($until ? {t_created => {'<=' => $until}} : undef, {order_by => ['me.id DESC']});
+sub latest_jobs ($self, $until = undef, $limit = ~0 >> 1) {
```

Comment:
Just make the default `undef` here. I suppose DBIx can cope with it. Otherwise I'd rather add the `limit` only to the hash if `$limit` is undef.

(Btw, `~0 >> 1` should be the highest possible integer (Perl's internal integer type can represent) divided by 2 (possibly rounded down). This might even be platform-dependent and doesn't look like a very meaningful default to pass to the database.)

Reply:
Yeah, I basically wanted "not specified". Mostly something to test it out with Tina's input. If we agree on an approach I'll see how to do it properly